src/services/webclient.ts

```diff
@@ -2590,12 +2590,20 @@ export class WebClientService {
             type: receiverType,
             id: receiverId,
         } as threema.Receiver;
-        this.messages.bindTemporaryToMessageId(
+        const message = this.messages.bindTemporaryToMessageId(
             receiver,
-            message.ack.id,
+            wireMessage.ack.id,
             messageId,
         );
         future.resolve(messageId);
+
+        // Add a special future that resolves once the message has been
+        // identified as sent.
```

Comment:
Maybe expand this a bit:

```suggestion
        // identified as sent. As long as an unacknowledged wire message future exists, the app will be woken up again on connection loss.
```

(With proper line breaks of course.)